### PR TITLE
Let server decide default repo visibility

### PR DESCRIPTION
**Pull request by Wauplin**
Until now, all repos were public by default when created without passing the `private` argument. This meant that passing `private=False` or `private=None` was strictly the same. This is not the case anymore. Enterprise Hub offers organizations to set a default visibility setting for new repos. This is useful for organizations forbidding public repos for security matters. This PR mostly updates some docstrings + default values so that `private=None` is always passed when user don't set it manually.

This PR doesn't create any breaking change. The real update has been done server-side when introducing the new Enterprise Hub feature. Related to https://github.com/huggingface/huggingface_hub/pull/2679.

### CODE REVIEW

File: src/diffusers/utils/hub_utils.py [hunk 1]
```diff
@@ -564,7 +564,8 @@ def push_to_hub(
             commit_message (`str`, *optional*):
                 Message to commit while pushing. Default to `"Upload {object}"`.
             private (`bool`, *optional*):
-                Whether or not the repository created should be private.
+                Whether to make the repo private. If `None` (default), the repo will be public unless the
+                organization's default is private. This value is ignored if the repo already exists.
```
Comment (pcuenca) [inline src/diffusers/utils/hub_utils.py:568]: Maybe link to the appropriate docs for enterprise orgs?

**Review by pcuenca — APPROVED** [review]

File: src/diffusers/utils/hub_utils.py (hunk 1 above)
Comment (Wauplin) [reply src/diffusers/utils/hub_utils.py:568]: I'd rather keep it simple (+ a bit lazy to update the 5 other PRs I've opened on other HF libraries :smile:)

**Comment by Wauplin** [comment]
(I cannot merge myself btw so I'll let a diffusers maintainer do it)

**Comment by yiyixuxu** [comment]
thanks @Wauplin !

**Pull request merged**